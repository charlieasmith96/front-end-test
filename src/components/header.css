.nav-bar {
    display: flex;
    height: 100px;
    background-color: #12A5ED;
}

.nav-bar-text {
    font-size: 70px;
    color: white;
    margin-left: 20px;
    margin-top: 0px
}

.nav-bar-logout {
    position: absolute;
    font-size: 35px;
    color: white;
    margin-top: 12px;
    right: 20px;
    padding: 10px;
    background: none;
	border: none;
	padding: 0;
	outline: none;
}