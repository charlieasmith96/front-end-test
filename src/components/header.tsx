import React, { useContext } from 'react';
import './header.css';
import { AuthContext } from '../authentication/auth-context-provider';

export const Header = () => {
  const { logout, isAuthenticated } = useContext(AuthContext);

  return (
    <div className="nav-bar">
      <div className="page-container">
        <h1 className="nav-bar-text">Data Dashboard</h1>
        {isAuthenticated && 
            <button data-testid="logout-button" className="nav-bar-logout" onClick={() => logout()}>Log out</button>}
      </div>
    </div>
  );
};
