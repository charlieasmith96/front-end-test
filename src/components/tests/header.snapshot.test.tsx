import React from 'react';
import renderer from 'react-test-renderer'
import { Header } from '../header';

it('should render header correctly', () => {
    const tree = renderer.create(<Header/>)
    expect(tree).toMatchSnapshot();
})