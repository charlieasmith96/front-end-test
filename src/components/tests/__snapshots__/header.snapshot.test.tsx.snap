// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders chart correctly 1`] = `
<div
  className="nav-bar"
>
  <div
    className="page-container"
  >
    <h1
      className="nav-bar-text"
    >
      Data Dashboard
    </h1>
  </div>
</div>
`;

exports[`should render header correctly 1`] = `
<div
  className="nav-bar"
>
  <div
    className="page-container"
  >
    <h1
      className="nav-bar-text"
    >
      Data Dashboard
    </h1>
  </div>
</div>
`;
